kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternDetailsController.java

```diff
@@ -208,36 +212,52 @@ private void initialize() {
-        otherNameText.textProperty().bind(otTextProperty);
+
+        // This will listen to the pattern descriptions event. Adding an FQN, Adding other name.
         patternDescriptionEventSubscriber = evt -> {
+            // This if is invoked when the data is coming from FQN name screen.
             if (evt.getEventType() == PatternDescriptionEvent.PATTERN_ADD_FQN) {
-                patternViewModel.setFullyQualifiedName(evt.getDescrName());
-                addFqnMenuItem.setVisible(fqnTextProperty.isEmpty().get());
+                DescrName descrName = evt.getDescrName();
+                patternViewModel.setPropertyValue(FQN_DESCRIPTION_NAME_TEXT, descrName.getNameText());
+                patternViewModel.setPropertyValue(FQN_DESCRIPTION_NAME, descrName);
             }
+
+            // This if is invoked when the data is coming from Other name name screen.
             if (evt.getEventType() == PatternDescriptionEvent.PATTERN_ADD_OTHER_NAME) {
-                patternViewModel.setOtherNameText(evt.getDescrName());
-            }
-            String dateAddedStr = LocalDate.now().format(DateTimeFormatter.ofPattern("MMM d, yyyy")).toString();
-            if (patternViewModel.getProperty(FQN_DESCRIPTION_NAME_TEXT) != null
-                && !((String)patternViewModel.getPropertyValue(FQN_DESCRIPTION_NAME_TEXT)).isEmpty()) {
-                fqnDate.setText("Date Added: " + dateAddedStr);
-                fqnDate.getStyleClass().add("text-noto-sans-normal-grey-eight");
-            } else {
-                fqnDate.setText("");
-            }
-            if (patternViewModel.getProperty(OTHER_NAME_DESCRIPTION_NAME_TEXT) != null
-                && !((String)patternViewModel.getPropertyValue(OTHER_NAME_DESCRIPTION_NAME_TEXT)).isEmpty()) {
-                otherNameDate.setText("Date Added: " + dateAddedStr);
-                otherNameDate.getStyleClass().add("text-noto-sans-normal-grey-eight");
-            } else {
-                otherNameDate.setText("");
+                ObservableList<DescrName> descrNameObservableList = patternViewModel.getObservableList(OTHER_NAMES);
+                descrNameObservableList.add(evt.getDescrName());
             }
         };
+
         EvtBusFactory.getDefaultEvtBus().subscribe(patternViewModel.getPropertyValue(PATTERN_TOPIC), PatternDescriptionEvent.class, patternDescriptionEventSubscriber);
 
+        ObjectProperty<DescrName> fqnNameProp = patternViewModel.getProperty(FQN_DESCRIPTION_NAME);
+        fqnNameProp.addListener((observableValue, oldDescrName, newDescrName) ->  {
+            fqnDescriptionSemanticText.setText(" (%s)".formatted(generateDescriptionSemantics(newDescrName)));
+            String dateAddedStr = newDescrName.getNameText().isEmpty()? "": LocalDate.now().format(DateTimeFormatter.ofPattern("MMM d, yyyy")).toString();
+            fqnAddDateLabel.setText(dateAddedStr);
+        });
```

Comment:
Not sure what error you were getting. 

My suggestion was off the top of my head. 
Here's my validated code that should work.
```java
// obtain reference of property (fqn descr name).
ObjectProperty<DescrName> objectProperty = patternViewModel.getProperty(FQN_DESCRIPTION_NAME);

// create a custom binding based on existence of FQN
StringBinding dateStrProp = Bindings
     .when(objectProperty.isNotNull())
     .then(LocalDate.now().format(DateTimeFormatter.ofPattern("MMM d, yyyy")))
     .otherwise("");

// target text property binds to custom binding (date or empty string)     
fqnAddDateLabel.textProperty().bind(dateStrProp);

```

Reply:
fqnAddDateLabel.textProperty().bind(fqnNameProp.map((fqnName) -> LocalDate.now().format(DateTimeFormatter.ofPattern("MMM d, yyyy"))).orElse(""));

Let me know if you still want to use the above logic.?